src/webgpu/gpu_test.ts

```diff
@@ -225,4 +233,79 @@ export class GPUTest extends Fixture {
     }
     return undefined;
   }
+
+  expectSingleColor(
+    src: GPUTexture,
+    format: GPUTextureFormat,
+    size: [number, number, number],
+    exp: PerTexelComponent<number>
+  ): void;
+  expectSingleColor(
+    src: GPUTexture,
+    format: GPUTextureFormat,
+    size: [number, number, number],
+    dimension: GPUTextureDimension,
+    exp: PerTexelComponent<number>
+  ): void;
+  expectSingleColor(
+    src: GPUTexture,
+    format: GPUTextureFormat,
+    size: [number, number, number],
+    dimension: GPUTextureDimension,
+    options: ExpectSingleColorOptions,
+    exp: PerTexelComponent<number>
+  ): void;
+  expectSingleColor(
+    src: GPUTexture,
+    format: GPUTextureFormat,
+    size: [number, number, number],
```

Comment:
getTextureCopyLayout uses this as the base (level 0) size, but copyTextureToBuffer uses it as the non-base size